node/src/components/storage.rs

```diff
@@ -1153,6 +1180,35 @@ fn insert_to_deploy_index(
     Ok(())
 }
 
+fn should_move_storage_files_to_network_subdir(root: &PathBuf, file_names: &[&str]) -> bool {
+    file_names
+        .iter()
+        .map(|file_name| root.join(file_name))
+        .all(|file_path| file_path.exists())
+}
+
+fn move_storage_files_to_network_subdir(
+    root: &PathBuf,
+    subdir: &PathBuf,
+    file_names: &[&str],
+) -> Result<(), Error> {
+    file_names
+        .iter()
+        .map(|file_name| (root.join(file_name), subdir.join(file_name)))
```

Comment:
I think rather than calling `map()` twice, it might be simpler and more readable to just do e.g.

```rs
        .map(|file_name| {
            let source_path = root.join(file_name);
            let dest_path = subdir.join(file_name);
            fs::rename(&source_path, &dest_path).map_err(|original_error| Error::UnableToMoveFile {
```
Also note that (with or without the change suggested above) we can avoid cloning the `PathBuf`s by passing refs in `fs::rename()`.

Reply:
Ohh for some reason I thought it had to take ownership. I didn't know it could borrow